concordium-smart-contract-testing/src/lib.rs

```diff
@@ -0,0 +1,2745 @@
+                                        new_balance: instance.self_balance,
+                                        data:        None,
+                                    }
+                                }
+                            };
+                            let success = matches!(response, InvokeResponse::Success { .. });
+                            // Add resume event
+                            self.chain_events.push(ChainEvent::Resumed {
+                                address: self.address,
+                                success,
+                            });
+
+                            let resume_res = v1::resume_receive(
+                                config,
+                                response,
+                                InterpreterEnergy::from(remaining_energy),
+                                &mut self.mutable_state,
+                                false, // never changes on transfers
+                                self.loader,
+                            );
+
+                            // Resume
+                            self.process(resume_res)
+                        }
+                        v1::Interrupt::Call {
+                            address,
+                            parameter,
+                            name,
+                            amount,
+                        } => {
+                            // Add the interrupt event
+                            self.chain_events.push(interrupt_event);
+
+                            if state_changed {
+                                println!("\t\tState was changed. Saving prior to another call.");
+                                let mut collector = SizeCollector::default();
+                                let persistent_state =
+                                    self.mutable_state.freeze(&mut self.loader, &mut collector);
+                                // TODO: Charge for size of new state.
+                                self.chain.get_instance_mut(address)?.state = persistent_state;
+                            }
+
+                            println!(
+                                "\t\tCalling contract {}\n\t\t\twith parameter: {:?}",
+                                address, parameter
+                            );
+
+                            let res = self.chain.contract_update_aux(
+                                self.invoker,
+                                Address::Contract(self.address),
+                                address,
+                                name,
+                                parameter,
+                                amount,
+                                self.invoker_amount_reserved_for_nrg,
+                                Energy::from(remaining_energy),
+                                &mut self.chain_events,
+                            );
+
+                            let (success, response, energy_after_invoke, state_changed) = match res
```

Comment:
Could we not instead change the return type of `contract_update_aux` to just return a struct with these fields. It would also not need to panic on an impossible case.